backend/src/apiserver/storage/pipeline_store.go

```diff
@@ -506,11 +495,26 @@ func (s *PipelineStore) CreatePipelineVersion(v *model.PipelineVersion) (*model.
 		return nil, util.NewInternalServerError(err, "Failed to add version to pipeline version table: %v",
 			err.Error())
 	}
-	_, err = tx.Exec(pipelineSql, pipelineArgs...)
-	if err != nil {
-		tx.Rollback()
-		return nil, util.NewInternalServerError(err, "Failed to update pipeline default version id: %v",
-			err.Error())
+	if updatePipelineVersion {
+		pipelineSql, pipelineArgs, pipelineErr := sq.
+			Update("pipelines").
+			SetMap(sq.Eq{"DefaultVersionId": newPipelineVersion.UUID}).
+			Where(sq.Eq{"UUID": newPipelineVersion.PipelineId}).
+			ToSql()
+		if pipelineErr != nil {
+			return nil, util.NewInternalServerError(
```

Comment:
I'd suggest still putting the pipelineSql initialization outside the transaction, because
e.g. you forgot to call `tx.Rollback()` before return here.

We should keep logic inside transaction as minimal as possible.

Reply:
Good point! Updated and moved it back